File: packages/sdk-ts/src/sdkIdentity.ts
import { STAGEHAND_SDK_VERSION } from "./version.js";

const STAGEHAND_SDK_IDENTITY = {
  name: "stagehand-sdk-ts",
  language: "typescript",
  version: STAGEHAND_SDK_VERSION,
} as const;

export const STAGEHAND_SDK_CLIENT_INFO = {
  name: STAGEHAND_SDK_IDENTITY.name,
  version: STAGEHAND_SDK_IDENTITY.version,
} as const;

export const STAGEHAND_SESSION_METADATA = {
  stagehand: "true",
  stagehand_sdk_language: STAGEHAND_SDK_IDENTITY.language,
  stagehand_sdk_version: STAGEHAND_SDK_IDENTITY.version,
} as const;

File: packages/sdk-python/src/stagehand/_sdk_identity.py
from importlib.metadata import PackageNotFoundError, version

from ._generated.models import ImplementationInfo


def _resolve_sdk_version() -> str:
    try:
        return version("stagehand")
    except PackageNotFoundError:
        return "0.0.0-dev"


STAGEHAND_SDK_VERSION = _resolve_sdk_version()

STAGEHAND_SDK_CLIENT_INFO = ImplementationInfo(
    name="stagehand-sdk-python",
    version=STAGEHAND_SDK_VERSION,
)

STAGEHAND_SESSION_METADATA = {
    "stagehand": "true",
    "stagehand_sdk_language": "python",
    "stagehand_sdk_version": STAGEHAND_SDK_VERSION,
}
